http/http_client.hpp

```diff
@@ -363,37 +387,56 @@ class HttpClient : public std::enable_shared_from_this<HttpClient>
             BMCWEB_LOG_ERROR
                 << "Maximum number of retries reached for Subscriber:" << subId;
 
-            // Clear queue.
-            while (!requestDataQueue.empty())
-            {
-                requestDataQueue.pop_front();
-            }
-
-            BMCWEB_LOG_DEBUG << "Retry policy: " << retryPolicyAction;
             if (retryPolicyAction == "TerminateAfterRetries")
             {
+                // Clear queue.
+                while (!requestDataQueue.empty())
+                {
+                    requestDataQueue.pop_front();
+                }
+
                 state = ConnState::terminated;
                 // Remove the subscription
                 BMCWEB_LOG_DEBUG << "TerminateAfterRetries is set. retryCount: "
                                  << retryCount << " .Subscriber: " << subId
                                  << " deleted";
                 persistent_data::EventServiceStore::getInstance()
                     .removeSubscription(subId);
+                return;
             }
             if (retryPolicyAction == "SuspendRetries")
             {
+                // Clear queue.
+                while (!requestDataQueue.empty())
+                {
+                    requestDataQueue.pop_front();
+                }
+
                 state = ConnState::suspended;
                 BMCWEB_LOG_ERROR
                     << "SuspendRetries is set. retryCount: " << retryCount
                     << " .Subscriber: " << subId << "suspended";
+                return;
             }
+
             BMCWEB_LOG_ERROR << retryPolicyAction
-                             << " is set. Cleanup queued events and "
-                                "retrycount for subId: "
+                             << " is set. Cleanup the current event and "
+                                "reset retrycount for subId: "
                              << subId;
             // Reset the retrycount to zero so that client can try connecting
             // again if needed
             retryCount = 0;
+
+            // All retries failed, Lets remove data from queue
+            if (!requestDataQueue.empty())
+            {
+                requestDataQueue.pop_front();
```

Comment:
We are just cleaning the current event which failed with all retries. We will try sending the next event in the queue to the subscriber